From c6fc4a5525a5ae1df7dc6b174f25843694a394be Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 22 Oct 2024 18:15:54 +0200
Subject: [PATCH 1455/1695] drivers: stm32_cryp: remove registering to
 shared_resource driver

Remove registering of STM32 CRYP driver to shared_resource driver. The
latter is deprecated since integration of the firewall framework
and will soon be removed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 84eec8535e6e0addab7c0fb5b5de2833d990eabd drivers: stm32_cryp: remove registering to shared_resource driver]
Change-Id: I8e40c006130a7187957b1b81dc7e171414cee614
Link: https://github.com/OP-TEE/optee_os/commit/67da2ad79d082db3c4f5cda1fef3213e0d531fc9
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/413519
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/crypto/stm32/stm32_cryp.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/core/drivers/crypto/stm32/stm32_cryp.c b/core/drivers/crypto/stm32/stm32_cryp.c
index 70ad4e38d..64d4c177a 100644
--- a/core/drivers/crypto/stm32/stm32_cryp.c
+++ b/core/drivers/crypto/stm32/stm32_cryp.c
@@ -1259,8 +1259,6 @@ static TEE_Result stm32_cryp_probe(const void *fdt, int node,
 	if (!cryp_pdata.base.va)
 		panic();
 
-	stm32mp_register_secure_periph_iomem(cryp_pdata.base.pa);
-
 	if (clk_enable(cryp_pdata.clock))
 		panic();
 
-- 
2.47.2

